### PR TITLE
Replaced the piece of source code for finding path with an existing function

### PR DESCRIPTION
**Pull request by wenqing**
as titled.

### CODE REVIEW

**Comment by wenqing** [comment]
@norihiro-w With  The path by pathDirname(FileName) misses the last path separator, and the existing function pathJoin() adds the separator. The last commit uses   pathJoin for every FilePath+[file name] for a correct file name with path.

File: FEM/rf_ic_new.cpp [hunk 1]
```diff
@@ -861,27 +861,7 @@ void CInitialCondition::SetDomain(int nidx)
...
-				funfname = FileName.substr(0, indexChWin);
-				funfname = funfname + "\\" + rfr_file_name;
-			}
-			else if (indexChLinux != string::npos)
-			{
-				funfname = FileName.substr(0, indexChLinux);
-				funfname = funfname + "/" + rfr_file_name;
-			}
-			restart_file_name = funfname;
+			std::string restart_file_name = FilePath + rfr_file_name;
```
Comment (norihiro-w) [inline FEM/rf_ic_new.cpp]: why not use `pathJoin()` in FileTools.h?

Comment (wenqing) [reply FEM/rf_ic_new.cpp]: I had a commit to replace FilePath + name with pathJoin but I removed it soon. With that replacement, "#include "FileTools"" were added in several files and many places were changed.  Later on I thought that using FilePath + name is more straightforward than calling pathJoin.

Comment (norihiro-w) [reply FEM/rf_ic_new.cpp]: i see

**Pull request merged**